windows/windows_update_install/prepare_msu_file.yml

```diff
@@ -41,26 +52,31 @@
 - name: "Copy the .msu file to local disk of guest OS"
   include_tasks: ../utils/win_execute_cmd.yml
   vars:
+    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
     win_powershell_cmd: >-
       {{ win_nfs_mount_cmd }};
-      Copy-Item -Path {{ msu_file_src_path }} -Include *.msu -Destination {{ msu_dir_path }}  -ErrorAction Stop;
+      $file = Get-ChildItem -Path {{ msu_file_nfs_path }} | where Name -like '{{ msu_file_name_reg }}';
+      if ($file) {Copy-Item -Path $file.FullName -Destination {{ msu_dest_dir }} -ErrorAction Stop;};
       net use {{ drive_letter_new }}: /delete
-    win_execute_cmd_no_log: "{{ hide_secrets | default(false) }}"
 
 - name: "Get the .msu file name"
   include_tasks: ../utils/win_execute_cmd.yml
   vars:
     win_powershell_cmd: >- 
-      Get-ChildItem -Path {{ msu_dir_path }} -Include *.msu -Name -ErrorAction Stop;
+      (Get-ChildItem -Path {{ msu_dest_dir }} | Where-Object {$_.name -like '*.msu'}).FullName;
 
-- name: "Check if the .msu file is copied to {{ msu_dir_path }}"
+- name: "Check if the .msu file is copied to {{ msu_dest_dir }}"
   ansible.builtin.assert:
     that:
       - win_powershell_cmd_output.stdout_lines is defined
       - win_powershell_cmd_output.stdout_lines | length != 0
-    fail_msg: "The .msu file is not found in {{ msu_dir_path }} in guest OS."
-    success_msg: "The .msu file is copied to {{ msu_dir_path }} in guest OS."
+    fail_msg: "The .msu file is not found in {{ msu_dest_dir }} in guest OS."
+    success_msg: "The .msu file is copied to {{ msu_dest_dir }} in guest OS."
+  when: msu_file_type == "LCU"
```

Comment:
Why does here only check LCU?

Reply:
LCU shoud be always offered, while SSU is not always offered.  So only add assert for LCU. 

Remind me that I can add a debug message to display the result of getting .msu file.